### PR TITLE
Create Database Structure - Game, User, Piece, Piece STIs

### PR DESCRIPTION
**Pull request by ruishine**

### CODE REVIEW

**Comment by ruishine** [comment]
@Methodologists/methodologists 
can I get a thumbs up! :)

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Piece < ActiveRecord::Base
   belongs_to :users
   belongs_to :games
+
+  def self.types
+    %w(Pawn Bishop Knight Rook Queen King)
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: What was the purpose? Any specific reasons to have it?

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class User < ActiveRecord::Base
+  has_many :games
+end
```
Comment (AlekstheRealist) [inline app/models/user.rb]: Shouldn't we add this?
`has_many :pieces`

File: app/models/piece.rb (hunk 1 above)
Comment (ruishine) [inline app/models/piece.rb]: Ah okay! I was following the STI tutorial and was confused by this line, so of course I just added it in for good measure.  I've deleted this in my new commit. Thank you!

File: app/models/user.rb (hunk 2 above)
Comment (ruishine) [inline app/models/user.rb]: You know what, I am still a little confused by the associations.  I've now added has_many :pieces in both game and user for good measure in the new commit.  Thanks!

**Pull request merged**